accounts/keystore/account_cache_test.go

```diff
@@ -0,0 +1,224 @@
+package keystore
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/accounts"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	cachetestDir, _   = filepath.Abs(filepath.Join("testdata", "keystore"))
+	cachetestAccounts = []accounts.Account{
+		{
+			Address: types.StringToAddress("7ef5a6135f1fd6a02593eedc869c6d41d934aef8"),
+		},
+		{
+			Address: types.StringToAddress("f466859ead1932d743d622cb74fc058882e8648a"),
+		},
+		{
+			Address: types.StringToAddress("289d485d9771714cce91d3393d764e1311907acc"),
+		},
+	}
+)
+
+func TestCacheInitialReload(t *testing.T) {
+	t.Parallel()
+
+	cache, _ := newAccountCache(cachetestDir, hclog.NewNullLogger())
+	accs := cache.accounts()
+
+	require.Equal(t, 3, len(accs))
+
+	for _, acc := range cachetestAccounts {
+		require.True(t, cache.hasAddress(acc.Address))
+	}
+
+	unwantedAccount := accounts.Account{Address: types.StringToAddress("2cac1adea150210703ba75ed097ddfe24e14f213")}
+
+	require.False(t, cache.hasAddress(unwantedAccount.Address))
+}
+
+func TestCacheAddDelete(t *testing.T) {
+	t.Parallel()
+
+	tDir := t.TempDir()
+
+	cache, _ := newAccountCache(tDir, hclog.NewNullLogger())
+
+	accs := []accounts.Account{
+		{
+			Address: types.StringToAddress("095e7baea6a6c7c4c2dfeb977efac326af552d87"),
+		},
+		{
+			Address: types.StringToAddress("2cac1adea150210703ba75ed097ddfe24e14f213"),
+		},
+		{
+			Address: types.StringToAddress("8bda78331c916a08481428e4b07c96d3e916d165"),
+		},
+		{
+			Address: types.StringToAddress("d49ff4eeb0b2686ed89c0fc0f2b6ea533ddbbd5e"),
+		},
+		{
+			Address: types.StringToAddress("7ef5a6135f1fd6a02593eedc869c6d41d934aef8"),
+		},
+		{
+			Address: types.StringToAddress("f466859ead1932d743d622cb74fc058882e8648a"),
+		},
+		{
+			Address: types.StringToAddress("289d485d9771714cce91d3393d764e1311907acc"),
+		},
+	}
+
+	for _, a := range accs {
+		require.NoError(t, cache.add(a, encryptedKeyJSONV3{}))
+	}
+	// Add some of them twice to check that they don't get reinserted.
+	require.Error(t, cache.add(accs[0], encryptedKeyJSONV3{}))
+	require.Error(t, cache.add(accs[2], encryptedKeyJSONV3{}))
+
+	// Check that the account list is sorted by filename.
+	wantAccounts := make([]accounts.Account, len(accs))
+
+	copy(wantAccounts, accs)
```

Comment:
It might be useful to add comment why we need this and to move these lines before deleting loop. That way is more obvious why we need them.